release/release.go

```diff
@@ -111,36 +110,53 @@ var fileExists = func(path string) bool {
 	return err == nil
 }
 
-var filesystemRootType = func() (int64, error) {
-	var statfs syscall.Statfs_t
-	if err := syscall.Statfs("/", &statfs); err != nil {
-		return 0, fmt.Errorf("cannot statfs filesystem root")
+// filesystemRootType returns the filesystem type mounted at "/".
+var filesystemRootType = func() (string, error) {
+	// We scan /proc/mounts, which contains space-separated values:
+	// [irrelevant] [mount point] [fstype] [irrelevant...]
+	// Here are some examples on some platforms:
+	// WSL1       :  rootfs / wslfs rw,noatime 0 0
+	// WSL2       :  /dev/sdc / ext4 rw,relatime,discard,errors=remount-ro,data=ordered 0 0
+	// lxc on WSL2:  /dev/loop0 / btrfs rw,relatime,idmapped,space_cache,user_subvol_rm_allowed,subvolid=259,subvol=/containers/testlxd 0 0
+	// We search for mount point = "/", and return the fstype.
+	file, err := os.Open("/proc/mounts")
```

Comment:
We have osutil.LoadMountInfo(); if we cannot use it (maybe it triggers a dependency cicle?) then please add a comment about it.

Reply:
Unfortunately it does cause a dependency cycle. I added the comment.